packages/valory/skills/transaction_settlement_abci/behaviours.py

```diff
@@ -538,10 +548,33 @@ def _check_tx_history(
             )
             return VerificationStatus.ERROR, None
 
+        contract_api_msg = yield from self._get_safe_nonce()
+        if (
+                contract_api_msg.performative != ContractApiMessage.Performative.STATE
+        ):  # pragma: nocover
+            self.context.logger.error(
+                f"get_safe_nonce unsuccessful! Received: {contract_api_msg}"
+            )
+            return VerificationStatus.ERROR, None
+
+        safe_nonce = cast(int, contract_api_msg.state.body["safe_nonce"])
+        if safe_nonce == self.params.mutable_params.nonce:
+            # if we have reached this state it means that the transaction didn't go through in the expected time
+            # as such we assume it is not verified
+            self.context.logger.info(
+                f"Safe nonce is the same as the nonce used in the transaction: {safe_nonce}. "
+                f"No transaction has gone through yet."
+            )
+            return VerificationStatus.NOT_VERIFIED, None
+
+
+        self.context.logger.info(
+            f"A transaction with nonce {safe_nonce} has already been sent. "
+        )
         self.context.logger.info(
-            f"Starting check for the transaction history: {self.history}."
+            f"Starting check for the transaction history: {self.history}. "
         )
-        for tx_hash in self.history[::-1]:
+        for tx_hash in self.history:
```

Comment:
we check even the current transaction, just to account for any issues that could've been in the prev. round

Reply:
Why not in reverse order? The logic here was that the more gas we spend in fees, the more probable for the tx to have passed.

Reply:
I see, that makes sense. I can revert this.